Sources/Workspace/UserToolchain.swift

```diff
@@ -35,12 +35,11 @@ public final class UserToolchain: Toolchain {
     /// The manifest resource provider.
     public let manifestResources: ManifestResourceProvider
 
+    public var toolPaths: [String: AbsolutePath]
```

Comment:
Doc comment and this property shouldn't be public. How about `toolsPathCache` as the name?

Reply:
done